File: src/python/01/solution.py

f = open("input.txt", "r")

elveList = f.read().split("\n\n")
print (len(elveList))
dictElveCaloriesCarried = {}

for elveNumber in range(len(elveList)):
    elveListItems = elveList[elveNumber].split("\n")
    elveCaloriesCarried = sum([int(x) for x in elveListItems])
    dictElveCaloriesCarried[elveNumber] = elveCaloriesCarried




print("Elve with max Calories: " + str(max(dictElveCaloriesCarried, key=dictElveCaloriesCarried.get)+1))
print("max Calories: " +   str(max(dictElveCaloriesCarried.values())))


caloriesList = list(dictElveCaloriesCarried.values())
caloriesList.sort(reverse=True)

caloriesByTopX = caloriesList
caloriesByTopX
for x in range(1, len(caloriesList)):
    caloriesByTopX[x] = caloriesByTopX[x-1] + caloriesList[x]

print("Calories carried by top 3 elves: " + str(caloriesByTopX[3-1]))

File: src/golang/01/solution.go
package main

import (
	"fmt"
	"os"
	"sort"
	"strconv"
	"strings"
)

func main() {
	b, err := os.ReadFile("input.txt")

	if err != nil {
		fmt.Print(err)
	}

	// fmt.Println(b)
	textFileContent := string(b)
	//fmt.Println(textFileContent)

	elveList := strings.Split(textFileContent, "\n\n")
	fmt.Println("Number of elves carrying stuff: ", len(elveList))

	caloriesCarried := []int{}

	for _, s := range elveList {
		caloriesCarried = append(caloriesCarried, sum(strings.Split(s, "\n")))
	}

	sort.Sort(
		sort.Reverse(
			sort.IntSlice(caloriesCarried),
		),
	)

	fmt.Println("Max Calories carried by one elve: ", caloriesCarried[0])

	carriedByXStrongest := caloriesCarried
	for i := 1; i < len(caloriesCarried); i++ {
		carriedByXStrongest[i] = carriedByXStrongest[i-1] + carriedByXStrongest[i]
	}

	fmt.Println("Calories carried by 3 strongest elves: ", carriedByXStrongest[3-1])
}

func sum(array []string) int {
	result := 0
	for _, v := range array {
		intVar, err := strconv.Atoi(v)
		if err != nil {
			//fmt.Print(err)
		}
		result += intVar
	}
	return result
}
